File: src/main/java/com/eager/questioncloud/workspace/WorkSpaceQuestionRegister.java
package com.eager.questioncloud.workspace;

import com.eager.questioncloud.question.Question;
import com.eager.questioncloud.question.QuestionContent;
import com.eager.questioncloud.question.QuestionRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class WorkSpaceQuestionRegister {
    private final QuestionRepository questionRepository;

    public Question register(Long creatorId, QuestionContent questionContent) {
        Question question = Question.create(creatorId, questionContent);
        return questionRepository.save(question);
    }
}

File: qc-core/qc-workspace/qc-workspace-api/src/main/kotlin/com/eager/questioncloud/workspace/implement/WorkspaceQuestionRegister.kt
package com.eager.questioncloud.workspace.implement

import com.eager.questioncloud.question.api.internal.QuestionCommandAPI
import com.eager.questioncloud.question.api.internal.RegisterQuestionAPIRequest
import com.eager.questioncloud.workspace.command.RegisterQuestionCommand
import org.springframework.stereotype.Component

@Component
class WorkspaceQuestionRegister(
    private val questionCommandAPI: QuestionCommandAPI,
) {
    fun registerQuestion(command: RegisterQuestionCommand) {
        questionCommandAPI.register(
            RegisterQuestionAPIRequest(
                command.creatorId,
                command.questionCategoryId,
                command.subject,
                command.title,
                command.description,
                command.thumbnail,
                command.fileUrl,
                command.explanationUrl,
                command.questionLevel,
                command.price,
            )
        )
    }
}